eclipse-theia/theia
# First cut of dynamic tabs with tab width assignment

#### What it does
Introduces a new, optional tab layout strategy similar to Chrome tabs as described in #12328 
This is part one, introducing only the dynamic tab resizing.

#### How to test
1. Turn on the preference "Workbench>Tab>DynamicTabs"
2. Open and close editors, observing the tab resizing behaviour
3. Make sure the tabs resizing works correctly when resizing
4. Make sure other tab usages (like the vertical tabs on the view dock panels) work fine and are unaffected by the dynamic tabs preference
5. Make sure the tab resizing works correctly when editors have different toolbar actions
6. Play with the "Workbench>Tab>Minimum Size | Preferred Size" settings and make sure the tab resizing works as expected

#### Review checklist


#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)
